cpp/scripts/gdb-pretty-printers.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import gdb
+
+
+if "HostIterator" not in locals():
```

Comment:
Make an explicit list of the names this relies on.
```suggestion
if not all(name in locals() for name in ("HostIterator", "DeviceIterator", "is_template_type_not_alias", "template_match")):
```